action.yml

```diff
@@ -22,84 +22,5 @@ inputs:
-          sudo chmod +x /usr/local/bin/docker-compose
-        fi
-
-        #!/bin/bash
-        set -e
-
-        max_retries=${{ inputs.max-retries }}
-        retry_interval=${{ inputs.retry-interval }}
-        compose_file=${{ inputs.compose-file }}
-        skip_exited=${{ inputs.skip-exited }}
-        skip_no_healthcheck=${{ inputs.skip-no-healthcheck }}
-
-        echo "Starting Docker Compose Health Check"
-        echo "Max retries: $max_retries"
-        echo "Retry interval: $retry_interval seconds"
-        echo "Compose file: $compose_file"
-        echo "Skip exited containers: $skip_exited"
-        echo "Skip containers without health checks: $skip_no_healthcheck"
-
-        check_services() {
-          if ! docker-compose -f "$compose_file" ps --services | grep -q .; then
-            echo "No services found"
-            return 1
-          fi
-
-          local not_ready=false
-          while read -r service; do
-            local container_id=$(docker-compose -f "$compose_file" ps -q $service)
-            local status=$(docker inspect --format='{{.State.Status}}' "$container_id")
-            local health=$(docker inspect --format='{{if .State.Health}}{{.State.Health.Status}}{{else}}N/A{{end}}' "$container_id")
-            
-            echo "Checking service $service: Status=$status, Health=$health"
-            
-            if [[ "$status" == "exited" && "$skip_exited" == "true" ]]; then
-              echo "Skipping exited container $service"
-              continue
-            fi
-
-            if [[ "$health" == "N/A" && "$skip_no_healthcheck" == "true" ]]; then
-              echo "Skipping container $service without health check"
-              continue
-            fi
-
-            if [[ "$status" != "running" ]] || [[ "$health" != "healthy" && "$health" != "N/A" ]]; then
-              echo "Service $service is not ready. Status: $status, Health: $health"
-              not_ready=true
-            fi
-          done < <(docker-compose -f "$compose_file" ps --services)
-
-          if $not_ready; then
-            return 1
-          else
-            return 0
-          fi
-        }
-
-        for i in $(seq 1 $max_retries); do
-          echo "Attempt $i of $max_retries"
-          if ! check_services; then
-            echo "Waiting for services to be ready... (Attempt $i/$max_retries)"
-            sleep $retry_interval
-          else
-            echo "All services are ready!"
-            docker-compose -f "$compose_file" ps
-            exit 0
-          fi
-          if [ $i -eq $max_retries ]; then
-            echo "Services did not become ready within the allocated time"
-            docker-compose -f "$compose_file" ps
-            docker-compose -f "$compose_file" logs
-            exit 1
-          fi
-        done
-      shell: bash
+  using: "node20"
```

Comment:
isnt there a version for node20?

Reply:
updated to use actions/setup-node@v4
https://github.com/jaracogmbh/docker-compose-health-check-action/pull/2/commits/3160ae5d77ee55c944e53cca9c567a88e314447e